### PR TITLE
Fix missing GC events on non-Windows for nativeaot

### LINKED ISSUE #87445: [NativeAOT] GC events don't get logged on Linux
The event firing in Native AOT is under a mix of `FEATURE_EVENT_TRACE` and `FEATURE_ETW`.

Some things don't even attempt to fire:
https://github.com/dotnet/runtime/blob/eb650227be048a1a89d18484de18dd5aeda7caf7/src/coreclr/nativeaot/Runtime/gctoclreventsink.cpp#L30-L37

Implementation of `ETW::GCLog` is in [eventtrace.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/Runtime/eventtrace.cpp), which is only included on Windows:
https://github.com/dotnet/runtime/blob/c32bee11a457151e5c400107ababed54dcd70438/src/coreclr/nativeaot/Runtime/CMakeLists.txt#L110-L115
(There's a bunch of stuff in there, so it's not quite as simple as including it on non-Windows)

Related (but more involved than): https://github.com/dotnet/runtime/issues/87325

cc @LakshanF

### PR DESCRIPTION
**Pull request by elinor-fung**
- Add a function for checking whether a keyword/level is enabled for the DotNETRuntime provider
- Include eventtrace.cpp based on `FEATURE_EVENT_TRACE` instead of only for Windows
- Separate eventtrace.cpp into different files based on functionality
  - All the code in eventtrace_bulktype.cpp and eventtrace_gcheap.cpp was directly from either eventtrace.cpp or rheventtrace.cpp
  - The bulktype and gcheap files remain Windows-only as we aren't prioritizing them yet
    - bulktype directly uses ETW functions
    - gcheap handles GCHeapDump and GCHeapSurvivalAndMovement - also, [WalkMovedReferences](https://github.com/dotnet/runtime/blob/e80ef8638b6d3f42add1bb046e032b2eff417473/src/coreclr/nativeaot/Runtime/gcrhenv.cpp#L960-L972) isn't hooked up, so I don't think GCHeapSurvivalAndMovement works

This does not change the ETW emitting, so that is still:

- Part of the core runtime 
  - The implementation for all that is right in header files, so it's not so simple to split out into the disabled/enabled eventing libs. We'll probably need to do it in order to do https://github.com/dotnet/runtime/issues/88162 in a reasonable way.
- Using the .NET Native provider GUIDs (47c3ba0c-77f1-4eb0-8d4d-aef447f16a85, private: 1095638c-8748-4c7a-b39e-baea27b9c589) instead of the normal .NET ones (e13c0d23-ccbc-4e12-931b-d9cc2eee27e4, private: 763FD754-7086-4dfe-95EB-C01A46FAF4CA).

Fixes https://github.com/dotnet/runtime/issues/87445

### CODE REVIEW

**Comment by jkotas** [comment]
> Separate eventtrace.cpp into different files based on functionality

nativeaot\Runtime\eventtrace.cpp and vm\eventtrace.cpp are same files. If you are splitting the nativeaot one into multiple files, you may consider doing the same for coreclr one too so that it is easier to compare and eventually unify.

File: src/coreclr/nativeaot/Runtime/rheventtrace.h [hunk 1]
```diff
@@ -5,20 +5,10 @@
 // This header provides Redhawk-specific ETW code and macros, to allow sharing of common
 // ETW code between Redhawk and desktop CLR.
 //
-#ifndef __RHEVENTTRACE_INCLUDED
-#define __RHEVENTTRACE_INCLUDED
+#ifndef RHEVENTTRACE_H
```
Comment (LakshanF) [inline src/coreclr/nativeaot/Runtime/rheventtrace.h:8]: Should this header file also be renamed?

File: src/coreclr/vm/eventtrace.cpp
Comment (LakshanF) [inline src/coreclr/vm/eventtrace.cpp:1]: Do we need FEATURE_NATIVEAOT definitions in this file?

File: src/coreclr/nativeaot/Runtime/eventtrace.cpp
Comment (LakshanF) [inline src/coreclr/nativeaot/Runtime/eventtrace.cpp:1]: This is not related to this change, but not sure of the design behind how GC events flow. `FireGCStart_V2` goes through here, `FireGCEnd_V1` skips `eventrace.cpp`. `RestartEE ` event takes another path

File: src/coreclr/nativeaot/Runtime/CMakeLists.txt [hunk 2]
```diff
@@ -222,6 +220,18 @@ endif()
 
 if(FEATURE_EVENT_TRACE)
   add_definitions(-DFEATURE_EVENT_TRACE)
+  list(APPEND COMMON_RUNTIME_SOURCES
```
Comment (LakshanF) [inline src/coreclr/nativeaot/Runtime/CMakeLists.txt:223]: I assume that Etw events would still be on by default, as opposed to opt in via `EventSourceSupport `flag.

File: src/coreclr/vm/eventtrace.cpp
Comment (elinor-fung) [reply src/coreclr/vm/eventtrace.cpp:1]: We don't. They should be gone now (removed in a different change, so they were probably still showing up before that went in and I rebased).

File: src/coreclr/nativeaot/Runtime/CMakeLists.txt (hunk 2 above)
Comment (elinor-fung) [reply src/coreclr/nativeaot/Runtime/CMakeLists.txt:223]: Yes, they remain on by default with this change.

File: src/coreclr/nativeaot/Runtime/eventtrace.cpp
Comment (elinor-fung) [reply src/coreclr/nativeaot/Runtime/eventtrace.cpp:1]: Looking at coreclr, I expect anything is more complicated (for example, requires extra work to get the information to fire, requires tracking additional information) goes through `ETW::*Log`, while the simple cases can directly `FireEtw*`

**Review by LakshanF — APPROVED** [review]

**Pull request merged**